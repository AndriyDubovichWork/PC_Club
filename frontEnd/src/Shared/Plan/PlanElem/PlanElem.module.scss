.PlanElem {
  width: 20vw;

  .title {
    font-size: clamp(1rem, 2vw, 3rem);
    font-weight: 700;
  }
  .subTitle {
    font-size: clamp(0.8rem, 2vw, 1rem);
    font-weight: 400;
  }
}
