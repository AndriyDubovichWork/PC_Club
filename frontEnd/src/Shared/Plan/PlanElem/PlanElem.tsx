import React from 'react';
import style from './PlanElem.module.scss';
import { ReactJSXElement } from '@emotion/react/types/jsx-namespace';
type PlanElemPropsT = {
  title: string;
  subTitle: string;
  children: ReactJSXElement;
};

const PlanElem: React.FC<PlanElemPropsT> = ({ title, subTitle, children }) => {
  return (
    <div className={style.PlanElem}>
      {children}
      <h1 className={style.title}>{title}</h1>
      <h3 className={style.subTitle}>{subTitle}</h3>
    </div>
  );
};

export default PlanElem;
