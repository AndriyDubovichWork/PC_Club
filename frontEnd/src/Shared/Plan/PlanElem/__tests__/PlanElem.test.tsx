import { render, screen, cleanup } from '@testing-library/react';
import PlanElem from '../PlanElem';
import renderer from 'react-test-renderer';
import { MemoryRouter, Route, Routes } from 'react-router-dom';

afterEach(() => {
  cleanup();
});

test('PlanElem match snapshot', () => {
  const tree = renderer
    .create(
      <MemoryRouter initialEntries={['/PC_Club/']}>
        <Routes>
          <Route
            path='/PC_Club/'
            element={
              <PlanElem
                title='Contact Us'
                subTitle='Our knowledgeable and friendly operators are always ready to help with any questions or concerns'
              >
                <div />
              </PlanElem>
            }
          />
        </Routes>
      </MemoryRouter>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});

test('PlanElem contains text', () => {
  render(
    <MemoryRouter initialEntries={['/PC_Club/']}>
      <Routes>
        <Route
          path='/PC_Club/'
          element={
            <PlanElem title='Contact Us' subTitle='Our knowledgeable'>
              <div />
            </PlanElem>
          }
        />
      </Routes>
    </MemoryRouter>
  );

  expect(screen.getByText('Contact Us')).toBeInTheDocument();
  expect(screen.getByText('Our knowledgeable')).toBeInTheDocument();
});
