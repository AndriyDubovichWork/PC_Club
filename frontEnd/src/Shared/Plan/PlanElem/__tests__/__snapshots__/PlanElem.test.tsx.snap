// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PlanElem match snapshot 1`] = `
<div
  className="PlanElem"
>
  <div />
  <h1
    className="title"
  >
    Contact Us
  </h1>
  <h3
    className="subTitle"
  >
    Our knowledgeable and friendly operators are always ready to help with any questions or concerns
  </h3>
</div>
`;
